recipes/opencv/4.x/conanfile.py

```diff
@@ -284,6 +301,17 @@ def add_components(components):
                 if self.settings.os == "Linux":
                     self.cpp_info.components[conan_component].system_libs = ["dl", "m", "pthread", "rt"]
 
+                if self.settings.os == "Android":
+                    self.cpp_info.components[conan_component].includedirs = [
+                        os.path.join("sdk", "native", "jni", "include")]
+                    self.cpp_info.components[conan_component].system_libs += ["log"]
+                    if not self.options.shared:
+                        self.cpp_info.components[conan_component].libdirs.append(
+                            os.path.join("sdk", "native", "staticlibs", tools.to_android_abi(str(self.settings.arch))))
+                        # TODO this doesn't work for cmake target based projects
+                        # self.cpp_info.components[conan_component].libdirs += ["lib"]
+                        # self.cpp_info.components[conan_component].libs += ["cpufeatures", "tegra_hal"]
```

Comment:
I had to add cpufeatures as a dependency for opencv_core to get it in the correct link order. Maybe this is the problem here too?

Reply:
Currently I have no idea how to add `cpufeatures.a` (and `tegra_hal.a` when I enable CAROTINE) in a `conan style` way, yes I think there is a problem with link order, we need some kind of internal dependency to a non-target library. would appreciate any help on this!

Reply:
I simply added cpufeatures as a dependency to core and added `self.cpp_infos.components["cpufeatures"].libs=["cpufeatures"]`.

Reply:
Hm, I think I tried something similar and failed, but not sure. Have to re-check. I wonder, why do I not get a linker error for missing `cpufeatures` at all right now? I don't include it, maybe it is part of the NDK already? Anyway, I have to dig into this again.

Reply:
You would also need the "main" component to require the `cpufeatures`

Reply:
IIRC I only got linker errors for cpufeatures in static builds.

Reply:
Thanks for the hint, I added `cpufeatures` now to `opencv_core`, where it is used. Now the link order looks good.